src/algebra/algebra/spectrum.lean

```diff
@@ -68,6 +68,16 @@ variable {R}
 noncomputable def resolvent (a : A) (r : R) : A :=
 ring.inverse (algebra_map R A r - a)
 
+/-- The unit `1 - r⁻¹ • a` constructed from `r • 1 - a` when the latter is a unit. -/
+@[simps]
+def units.sub_inv_smul {r : Rˣ} {a : A}
+  {u : Aˣ} (h : (u : A) = r • 1 - a) : Aˣ :=
+{ val := 1 - r⁻¹ • a,
+  inv := r • ↑u⁻¹,
+  val_inv := by { rw [mul_smul_comm, ←smul_mul_assoc, smul_sub, smul_inv_smul, ←h],
+                  exact u.val_inv },
+  inv_val := by { rw [smul_mul_assoc, ←mul_smul_comm, smul_sub, smul_inv_smul, ←h],
+                  exact u.inv_val } }
```

Comment:
How about:
```suggestion
@[simps] noncomputable def units.sub_inv_smul {r : Rˣ} {a : A} (h : is_unit $ r • 1 - a) : Aˣ :=
{ val := 1 - r⁻¹ • a,
  inv := r • ↑h.unit⁻¹,
  val_inv := by rw [mul_smul_comm, ←smul_mul_assoc, smul_sub, smul_inv_smul, h.mul_coe_inv],
  inv_val := by rw [smul_mul_assoc, ←mul_smul_comm, smul_sub, smul_inv_smul, h.coe_inv_mul], }
```

Reply:
done, but also generalized slightly to accommodate the suggestion below.